### PR TITLE
mgmt compute support vm size enum generation

### PR DESCRIPTION
**Pull request by xseeseesee**
As compute service team mentioned before, they plan to remove the VM sizes from swagger spec. The recent [spec PR](https://github.com/Azure/azure-rest-api-specs/pull/11598) shows they already start to work for this.

To avoid breaking change from SDK side, this PR could be potential solution to list resource skus and generate latest VM size enum type when it's required to update. After this got merged, I can open another one to revise the fixed value we applied to vm tests before.

Update: 
The revapi reports below error
```
[ERROR] java.field.removed: field com.azure.resourcemanager.compute.models.VirtualMachineSizeTypes.STANDARD_E32_16_V3: Field removed from class. (breaks semantic versioning)
```
Actually the swagger is incorrect [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json#L7936).
The same issue already reported [here](https://github.com/MicrosoftDocs/azure-docs/issues/30378), while the fix only applied to docs.

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
Please take a note whether there is any sku type that get removed in this PR.

**Review by ChenTanyi — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
Add to CHANGELOG

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/resources/VirtualMachineSizeTypes.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+package com.azure.resourcemanager.compute.models;
+
+import com.azure.core.util.ExpandableStringEnum;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.Collection;
+
+/** Defines values for VirtualMachineSizeTypes. */
+public final class VirtualMachineSizeTypes extends ExpandableStringEnum<VirtualMachineSizeTypes> {
+TYPES_PLACE_HOLDER
```
Comment (JonathanGiles) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/resources/VirtualMachineSizeTypes.java:12]: This doesn't appear as if it would compile?

Comment (xccc-msft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/resources/VirtualMachineSizeTypes.java:12]: It's placed under `../src/test/resources/..` as template file. We prepared one test method to pull all the values and generate/update the enum class by this template.

**Comment by xccc-msft** [comment]
Close this as the spec PR changed. They plan to remove the enum data type on Dec 23rd, 2023 while now they add deprecate message instead of removing the enum values.

**Pull request closed without merging**